epictrack-api/src/api/resources/work_status.py

```diff
@@ -35,5 +36,56 @@ class WorkStatus(Resource):
     @auth.require
     def get(work_id):
         """Return all active works."""
-        works = WorkStatusService.find_all_works_tatus(work_id)
+        works = WorkStatusService.find_all_work_status(work_id)
         return jsonify(res.WorkStatusResponseSchema(many=True).dump(works)), HTTPStatus.OK
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def post(work_id):
+        """Create new work status"""
+        request_dict = req.WorkStatusParameterSchema().load(API.payload)
+        work_status = WorkStatusService.create_work_status(work_id, request_dict)
+        return res.WorkStatusResponseSchema().dump(work_status), HTTPStatus.CREATED
+
+
+@cors_preflight("GET, PUT")
+@API.route("/<int:status_id>", methods=["GET", "PUT", "OPTIONS"])
+class Status(Resource):
+    """Endpoint resource to manage a work status."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def put(work_id, status_id):
+        """Update work status"""
+        request_dict = req.WorkStatusParameterSchema().load(API.payload)
+        existing_work_status = WorkStatusService.find_work_status_by_id(work_id, status_id)
+        if existing_work_status is None:
```

Comment:
do we prefer having this logic at the resource layer rather than the service layer?